proto/build.rs

```diff
@@ -93,5 +93,4 @@ static FIELD_ATTRIBUTES: &[(&str, &str)] = &[
     (".penumbra.stake.IdentityKey.ik", AS_BECH32_IDENTITY_KEY),
     (".penumbra.crypto.Address.inner", AS_BECH32_ADDRESS),
     (".penumbra.crypto.AssetId.inner", AS_BECH32_ASSET_ID),
-    (".penumbra.crypto.NoteCommitment.inner", AS_HEX),
```

Comment:
Without the `AS_HEX`, this passes the roundtrip parse test, and `pcli` can sync the chain correctly, picking up where it left off by re-parsing the NCT appropriately. It's not immediately clear to me why adding `AS_HEX` here is causing a deserialization failure for bincode. Any ideas?

Reply:
I'm really unsure why this would cause a deserialization failure.  From the bytes you posted, it looked like the encoding was 8 bytes of length encoding "64", then 64 bytes of UTF-8 text with the hex encoding.  The error we got on deserialization occurs when bincode is trying to decode an `Option`; it errors because it's expecting 0 (`None`) or 1 (`Some`), but sees 64, the length field, and treats it as an enum discriminant.

One problem with this fix is that I think it will screw up the JSON format used for wallet.json; maybe that's less important than having it work, but it's something to consider.